### ISSUE TITLE
Remote gdb uses wrong architecture

### ISSUE THREAD

**Issue by aronsky** [radareorg/radare2#14987]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | OSX and Ubuntu, x86-64
| File format of the file you reverse (mandatory)      | None (remote gdb)
| Architecture/bits of the file (mandatory)            | BA2
| r2 -v full output, **not truncated** (mandatory)     | radare2 3.8.0 23032 @ darwin-x86-64 git.3.8.0 commit: b4860e4eecad2053202965926f16296864b2f1e5 build: 2019-09-08__16:30:11

### Expected behavior

When using remote gdb with my r2lang plugin for a custom architecture (https://github.com/alephsecurity/pyba2), I expect r2 to use the registers from the architecture. When opening a regular file, it works (in ESIL mode, for example), and the correct registers for the architecture are shown.

### Actual behavior

x86 registers are shown (even though the correct arch is selected, and disassembly listings are correct).

### Steps to reproduce the behavior
I'm using a custom build of QEMU with support for the architecture (it's a WIP) as the gdbserver, but I think it should work with any gdbserver.

Basically:
1. Create a custom architecture (possibly using r2lang, not sure if it's critical).
2. Connect to a remote gdb instance, and set the arch to your custom architecture.
3. Check the registers with `dr`.

**Comment by radare** [comment]
Do you use -a and -b to specify that, its not autodetected

**Comment by aronsky** [comment]
I do.
I found the problem, actually (haven't had the chance to update the issue, only posted it on Telegram).

The issue is that `r_debug_set_arch` calls `r_sys_arch_match` (https://github.com/radare/radare2/blob/fe9a034/libr/debug/debug.c#L463). That function looks at `dbg->h->arch`, which is populated at https://github.com/radare/radare2/blob/fe9a03416f646a8acc0eb4173bca354aa98907eb/libr/debug/p/debug_gdb.c#L1104 and set to `"x86,arm,sh,mips,avr,lm32,v850"`. Therefore, my arch (that's not one of those 7) doesn't match, and `r_debug_set_arch` fails.

As a temporary workaround, I just added my arch to that string, and it works great (there are issues with setting `bin.baddr`, but I'm working on it - it's at least partially the fault of QEMU's gdbstub).

However, I'm wondering what would be the correct and elegant way to fix it upstream. We could add `any` to the supported archs of `debug_gdb`, because basically as long as we connect to a gdbserver at the back, it's kind of supported. Or, we could add some support to update the list of supported archs programmatically?

**Comment by radare** [comment]
yeah, feel free to add another entry there for your atch, not sure if deleting the whitelist would be good at all

can you do a  PR? so we can discuss it in there?

**Comment by XVilka** [comment]
@aronsky was this fixed?

**Comment by aronsky** [comment]
Yes, sorry. There was a PR that was merged.
